File: src/test/java/dev/retrotv/crypt/owe/sha/SHA512224Test.java
package dev.retrotv.crypt.owe.sha;

import dev.retrotv.crypt.Algorithm;
import dev.retrotv.crypt.owe.OWETest;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class SHA512224Test extends OWETest {

    @Test
    @DisplayName("SHA512224 File hash 테스트")
    void fileHashTest() throws Exception {
        fileHashTest(Algorithm.SHA512224);
    }

    @Test
    @DisplayName("SHA512224 File hash matches 테스트")
    void fileHashMatchesTest() throws Exception {
        fileHashMatchesTest(new SHA512224(), Algorithm.SHA512224);
    }

    @Test
    @DisplayName("SHA512224 File and File matches 테스트")
    void fileMatchesTest() throws Exception {
        fileMatchesTest(new SHA512224());
    }

    @Test
    @DisplayName("SHA512224 password encode 테스트")
    void passwordEncrypt() {
        passwordEncryptAndMatchesTest(new SHA512224());
    }
}

File: src/test/kotlin/dev/retrotv/crypto/owe/hash/sha/SHA512224Test.kt
package dev.retrotv.crypto.owe.hash.sha

import dev.retrotv.crypto.owe.OWETest
import dev.retrotv.enums.Algorithm
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class SHA512224Test : OWETest() {
    @Test
    @DisplayName("SHA512224 File hash 테스트")
    fun fileHashTest() {
        fileHashTest(Algorithm.Hash.SHA512224)
    }

    @Test
    @DisplayName("SHA512224 File hash matches 테스트")
    fun fileHashMatchesTest() {
        fileHashMatchesTest(SHA512224(), Algorithm.Hash.SHA512224)
    }

    @Test
    @DisplayName("SHA512224 password encode 테스트")
    fun passwordEncrypt() {
        passwordEncryptAndMatchesTest(SHA512224())
    }
}
